endpoint/register/pypush_gsa_icloud.py

```diff
@@ -237,19 +233,24 @@ def sms_second_factor(dsid, idms_token):
 
     headers.update(generate_anisette_headers())
 
-    # TODO: Actually get the correct id, probably in the above GET
-    body = {"phoneNumber": {"id": 1}, "mode": "sms"}
+    # Extract the "boot_args" from the auth page to get the id of the trusted phone number
+    pattern = r'<script type="application/json" class="boot_args">\s*(.*?)\s*</script>'
+    auth = requests.get("https://gsa.apple.com/auth", headers=headers, verify=False)
+    match = re.search(pattern, auth.text, re.DOTALL)
+    boot_args = json.loads(match.group(1).strip())
+
+    body = {"phoneNumber": {"id": boot_args["direct"]["phoneNumberVerification"]["trustedPhoneNumber"]["id"]}, "mode": "sms"}
 
     # This will send the 2FA code to the user's phone over SMS
     # We don't care about the response, it's just some HTML with a form for entering the code
     # Easier to just use a text prompt
-    t = requests.put(
```

Comment:
@FDHoho007 curious how this is functional without making the put request here?

Reply:
Ok I see your comment about getting 2 requests. I got only one request when leaving this uncommented.
Otherwise, I don't see how else you would send the body you constructed with the correct 2fa phoneNumber id without this call.

Reply:
Yeah thats the funny thing. It seems calling https://gsa.apple.com/auth to get the phoneNumber id is already sufficient to request a 2FA code to your primary trusted device.
So in my case calling https://gsa.apple.com/auth replaced the need for https://gsa.apple.com/auth/verify/phone/. But as mentioned above I'm not certain, that this applies to all users and situations. Thats why explicitly left the code in there.
I'm currently waiting for some kind of suggestions or guidance which approach to choose here, since I am not at all familiar with the Apple APIs and just started using this project.